dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/ChatPromptParser.cs

```diff
@@ -48,18 +52,43 @@ private static bool TryParse(List<PromptNode> nodes, [NotNullWhen(true)] out Cha
     {
         chatHistory = null;
 
-        foreach (var node in nodes)
+        foreach (var node in nodes.Where(IsValidChatMessage))
         {
-            if (IsValidChatMessage(node))
-            {
-                var role = node.Attributes[RoleAttributeName];
-                var content = node.Content!;
+            (chatHistory ??= new()).Add(ParseChatNode(node));
+        }
+
+        return chatHistory is not null;
+    }
 
-                (chatHistory ??= new()).AddMessage(new AuthorRole(role), content);
+    /// <summary>
+    /// Parses a chat node and constructs a ChatMessageContent object.
+    /// </summary>
+    /// <param name="node">The prompt node to parse.</param>
+    /// <returns>A ChatMessageContent object.</returns>
+    private static ChatMessageContent ParseChatNode(PromptNode node)
+    {
+        ChatMessageContentItemCollection items = new();
+        foreach (var childNode in node.ChildNodes)
+        {
+            if (childNode.TagName.Equals(ImageTagName, StringComparison.OrdinalIgnoreCase) && childNode.Content is not null)
+            {
+                items.Add(new ImageContent(new Uri(childNode.Content)));
+            }
+            else if (childNode.TagName.Equals(TextTagName, StringComparison.OrdinalIgnoreCase) && childNode.Content is not null)
+            {
+                items.Add(new TextContent(childNode.Content));
             }
         }
 
-        return chatHistory is not null;
+        if (items.Count == 1 && items[0] is TextContent textContent)
+        {
+            node.Content = textContent.Text;
+            items.Clear();
+        }
+
+        return items.Count > 0
+            ? new ChatMessageContent(new AuthorRole(node.Attributes[RoleAttributeName]), items)
+            : new ChatMessageContent(new AuthorRole(node.Attributes[RoleAttributeName]), node.Content);
```

Comment:
Little simplification:
```suggestion
        var authorRole = new AuthorRole(node.Attributes[RoleAttributeName]);

        return items.Count > 0
            ? new ChatMessageContent(authorRole, items)
            : new ChatMessageContent(authorRole, node.Content);
```